dcos/mesos.py

```diff
@@ -1015,8 +1015,16 @@ def __init__(self, task_id, cmd=None, args=None,
         client = DCOSClient()
         master = get_master(client)
 
-        # Get the URL to the agent running the task.
+        # Get the task and make sure its a container launched by the UCR.
         task_obj = master.task(task_id)
+        if "container" in task_obj.dict():
+            if "type" in task_obj.dict()["container"]:
+                if task_obj.dict()["container"]["type"] != "MESOS":
```

Comment:
I had something similar to this at first, but the problem is I need to allow everything through except the one case where `task_obj.dict()["container"]["type"] != "MESOS":`. Meaning, I want to allow it to continue even `if "container" not in task_obj.dict()` and `if "type" in task_obj.dict()["container"]`.

I didn't see a better way of getting these semantics.